src/app/application.cpp

```diff
@@ -322,7 +322,7 @@ void Application::runExternalProgram(const BitTorrent::Torrent *torrent) const
     };
 #endif
 
-    QString program = Preferences::instance()->getAutoRunProgram().trimmed();
+    QString program = QString(targetProgram);
```

Comment:
???
```suggestion
    QString program = targetProgram;
```

Reply:
As the argument is constant (I did not want to deviate from the conventions, everything else that is an object is constant in argument lists across the project AFAIR), but _its value actually getting updated/changed in the method_ with the parameter substitution, I figured I'd copy it with the constructor as Qt documentation states.

The documentation for the assignment operator, to me, _seems_ to imply that _assignment is by reference_.
https://doc.qt.io/qt-5/qstring.html#operator-eq
**Please let me know if I'm wrong.**

Please note that I am new to Qt and have no knowledge about implementation details of QString copying, but have the habit to consider inadvertently changed arguments a safety issue.

Reply:
After another take, I verified that due to .trimmed(), there is not even marginal possibility overwriting the original setting (that could lead to passing a command line substituted with outdated data), and also that it compiles with gcc in the way you recommended, so I indeed changed that.
Now I await the CD results and hope that the clang environment (for macOS) that is known to have a somewhat more picky static analysis at times won't complain.

Reply:
>Please note that I am new to Qt and have no knowledge about implementation details of QString copying, but have the habit to consider inadvertently changed arguments a safety issue.
>
>EDIT: I checked the source for QString and it seems that the assignment operator is indeed by reference, and thus unsafe.

Seems you're new not only to Qt but to C++ as well.

Reply:
Well, given that my previous experience with Qt is literally zero, I have infinitely more experience with C++ than with Qt. :)
But at the end of the day you are right; that's the primary reason I embark on endeavours like this.